tracing/tracing.go

```diff
@@ -0,0 +1,82 @@
+package tracing
+
+import "context"
+
+type contextKey string
+
+const (
+	traceContextKey contextKey = "trace"
+)
+
+// TracerFromContext returns a tracer from the context. If no tracer is found,
+// an empty tracer is returned.
+func TracerFromContext(ctx context.Context) Tracer {
+	value := ctx.Value(traceContextKey)
+	if value == nil {
+		return noopTracer{}
+	}
+	tracer, ok := value.(Tracer)
+	if !ok {
+		return noopTracer{}
+	}
+	return tracer
+}
+
+// WithTracer returns a new context with the given tracer.
+func WithTracer(ctx context.Context, tracer Tracer) context.Context {
+	if tracer == nil {
+		tracer = noopTracer{}
+	}
+	return context.WithValue(ctx, traceContextKey, tracer)
+}
```

Comment:
Does this definition need to be in go-dqlite? My understanding is that creating a context with tracing powers is the responsibility of downstream, so it would make more sense to me to just expose the value of `traceContextKey` instead.

Reply:
The rationale behind not exposing the key and exposing the function is that gives you the option of changing the key or how the trace is added to the context. It's generally a [go pattern](https://medium.com/@matryer/context-keys-in-go-5312346a868d) to do it like this for context-additive operations.

I'm not wedded to this pattern if you do not want to expose this function, but instead offer the `TraceContextKey` (as public) then I'm happy to do that as well.

Reply:
No, that's fine, we should use the conventional Go style for this.